packages/tocco-ui/src/Button/StyledButton.js

```diff
@@ -0,0 +1,110 @@
+import styled from 'styled-components'
+import {theme} from 'styled-system'
+
+import {StyledButtonGroup} from '../ButtonGroup'
+import {
+  declareDensity,
+  declareElevation,
+  declareFlatBaseColors,
+  declareFlatPrimaryColors,
+  declareInteractionColors,
+  declareOverlay,
+  declareRaisedBaseColors,
+  declareRaisedPrimaryColors,
+  stylingInk,
+  stylingLook,
+  stylingPosition
+} from '../utilStyles'
+
+const meltButtons = props => {
+  let declaration = ''
+  if (!props.melt && props.look === stylingLook.RAISED) {
+    declaration = `
+      &:not(:last-child) {
+        margin-right: ${theme('space.4')(props)};
+      }
+    `
+  } else if (props.melt) {
+    declaration = `
+      border-radius: 0;
+
+      &:first-child {
+        border-top-left-radius: ${theme('radii.3')(props)};
+        border-bottom-left-radius: ${theme('radii.3')(props)};
+      }
+
+      &:last-child {
+        border-top-right-radius: ${theme('radii.3')(props)};
+        border-bottom-right-radius: ${theme('radii.3')(props)};
+      }
+    `
+  }
+  return declaration
+}
+
+const declareButtonColor = props => {
+  if (props.look === stylingLook.FLAT && props.ink === stylingInk.BASE) {
+    return declareInteractionColors(declareFlatBaseColors(props))
+  } else if (props.look === stylingLook.FLAT && props.ink === stylingInk.PRIMARY) {
+    return declareInteractionColors(declareFlatPrimaryColors(props))
+  } else if (props.look === stylingLook.RAISED && props.ink === stylingInk.BASE) {
+    return declareInteractionColors(declareRaisedBaseColors(props))
+  } else if (props.look === stylingLook.RAISED && props.ink === stylingInk.PRIMARY) {
+    return declareInteractionColors(declareRaisedPrimaryColors(props))
+  }
+}
+
+const declareIconPosition = props => {
+  if (props.iconPosition === stylingPosition.AFTER) {
+    return `
+      justify-content: space-between;
+      > span {
+        order: -1;
+      }
+    `
+  }
+}
+
+const StyledButton = styled.button`
+  && {
+    align-items: center;
+    background-image: none;
+    border-radius: ${theme('radii.3')};
+    border: none;
+    display: inline-flex;
+    margin: 0;
+    position: relative;
+    text-align: center;
+    text-transform: uppercase;
+    vertical-align: middle;
+    white-space: nowrap;
+
+    &:enabled {
+      cursor: pointer;
+    }
+
+    &:disabled {
+      ${props => declareOverlay(theme('overlays.disabled.color')(props), theme('overlays.disabled.opacity')(props))}
+    }
+
+    &:active,
+    &:focus {
+      outline: ${theme('outline')};
+    }
+
+    ${props => declareButtonColor(props)}
+    ${props => declareDensity(props)}
+    /*
+      SCR_TEMP reactivate
```

Comment:
Not a huge fan of TEMP code that is commented out. Please remove all SRC_TEMP an create a patch with the code. So in case we want to fully adapt material ui we can commit this patch.

Reply:
see 3fd9ef86227c66dd3c9ce11abae8e0a32e41d453